crates/consensus/src/manager.rs

```diff
@@ -160,10 +141,13 @@ where
         match new_stat {
             // means we transitioned from commit phase to bid submission.
             // nothing much to do here. we just wait sometime to accumulate orders
-            ConsensusState::BidSubmission(BidSubmission { pre_proposals, .. }) => {}
+            ConsensusState::PreProposalSubmission(PreProposalSubmission { .. }) => {}
             // means we transitioned from bid submission to aggregation, therefore we broadcast our
             // pre-proposal to the network
-            ConsensusState::BidAggregation(BidAggregation { pre_proposals, .. }) => {
+            ConsensusState::PreProposalAggregation(PreProposalAggregation {
```

Comment:
should we not start collecting pre-proposals here so when we move to pre-proposal aggregation that we can just instantly send ours out?

Reply:
`on_state_start()` is meant to be used as a place to trigger a side effect at the beginning of the new state, that is different to the side effect that is part of the state itself. 

When `PreProposalSubmission` starts we are in the 0 - 3 seconds "grace period" after the last block where we allow the current order storage to invalidate orders *.  At this point it is assumed that we are not ready to build a porposal because we not allowed the order_storage to flush itself.

(it was supposed to, but wasn't; has been fixed now) Preposals are being collected continuously in `on_strom_message` as longs as we are not in `Finalized`, because at that point the leader already submitted on chain. 


*The reason why that is there is because the order storage cleans itself after fork choice and this is just a sync mechanism. Really the better solution would be to create a channel between the storage and the manager so that the storage can tell the manager when it has finished cleaning , thus the manager can start building a proposal again.